sdk-lib/src/main/java/cash/z/ecc/android/sdk/model/Account.kt

```diff
@@ -0,0 +1,16 @@
+package cash.z.ecc.android.sdk.model
+
+/**
+ * A [ZIP 316](https://zips.z.cash/zip-0316) account identifier.
+ *
+ * @param value A 0-based account index.  Must be >= 0.
+ */
+data class Account(val value: Int) {
+    init {
+        require(value >= 0) { "Account index must be >= 0 but actually is $value" }
```

Comment:
An account must be between `0` and `2^31 - 1` inclusive (`2^31 - 1` is valid for derivation, but is used for legacy `zcashd` purposes, so we may want to exclude that as well for the purposes of the SDK).

Reply:
Kotlin Int is 32-bits.  Effectively it is a Java Integer.  So https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Integer.html#MAX_VALUE

Reply:
Okay, so restricting to positive is equivalent to restricting to a 31-bit range, great. The `2^31 - 1` exclusion may want to happen, but that doesn't have to block this PR.

Reply:
I filed a followup issue to come back to this: https://github.com/zcash/zcash-android-wallet-sdk/issues/729